packages/dds/map/src/mapKernel.ts

```diff
@@ -112,6 +111,40 @@ export interface IMapDataObjectSerialized {
     [key: string]: ISerializedValue;
 }
 
+interface IMapKeyEditLocalOpMetadata {
+    type: "edit";
+    pendingMessageId: number;
+    previousValue?: ILocalValue;
```

Comment:
It's not obvious that an undefined value for previousValue indicates the unknown case from stashed op, so it can't be rolled back. (If it's known that there was no previous value, IMapKeyAddLocalOpMetadata is used instead.) I'm slightly inclined to add another localOpMetadata type for this case.